src/app/core/advertiser/advertiser.service.ts

```diff
@@ -1,32 +1,86 @@
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
-import { HalDoc } from 'ngx-prx-styleguide';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { withLatestFrom, map, mergeMap, publish, refCount } from 'rxjs/operators';
 import { AuguryService } from '../augury.service';
+import { HalDoc } from 'ngx-prx-styleguide';
 
 export interface Advertiser {
   id: number;
   name: string;
-  self_uri: string;
+  set_advertiser_uri: string;
 }
 
 @Injectable()
 export class AdvertiserService {
-  constructor(private augury: AuguryService) {}
+  // tslint:disable-next-line: variable-name
+  private _advertisers = new BehaviorSubject<{[id: number]: Advertiser}>({});
+
+  constructor(private auguryService: AuguryService) {
+    this.loadAdvertisers();
+  }
 
-  listAdvertisers(params = {}): Observable<Advertiser[]> {
-    return this.augury.followItems('prx:advertisers', params).pipe(map(docs => this.docsToAdvertisers(docs)));
+  get advertisers(): Observable<Advertiser[]> {
+    return this._advertisers.pipe(
+      map(advertisers => {
+        return Object.keys(advertisers)
+          .map(id => advertisers[id])
+          .sort((a, b) => a.name.localeCompare(b.name));
+      })
+    );
   }
 
-  docsToAdvertisers(docs: HalDoc[]): Advertiser[] {
-    return docs.map(this.docToAdvertiser);
+  loadAdvertisers() {
+    this.auguryService.followItems(`prx:advertisers`, {per: 999}).pipe(
```

Comment:
But that wouldn't give us all the advertisers, just the campaign advertiser, right? (Also campaign already embeds the advertiser by default, but it seems like we only needed the uri on the campaign model that then maps to this list in order to show names on the form.)

Reply:
The campaign (+ advertiser) always loads first.  So to avoid the form flashing, I guess I'm saying we could pre-populate this store with just the 1 advertiser.  Then later, add _all_ the advertisers but dedup the one we already added (by id/uri).

Reply:
Seems like that could work. I'd kind of like to figure out why the form isn't just displaying the value reactively when it gets advertisers, seems like that should just work in this scenario if we set everything up.

Reply:
So I tried this, but I don't really like it, too messy. The campaign (api) service loads the single advertiser with the campaign, which it returns back to the campaign store then the campaign store has to update the advertiser store. If we _have_ to go this route, I guess but I am just not feeling it so far.

Reply:
I think what we should do instead of tangling up these stores is load the data sequentially to resolve the race condition. Load the accounts and advertisers, then load the campaign.

Reply:
I set it up to load advertisers then campaign. I left accounts alone for now just trying to focus on one thing at a time.

Reply:
+1